src/components/ShoppingControls/ShoppingControls.js

```diff
@@ -0,0 +1,81 @@
+import React, { useContext, useState } from "react";
+import { ShoppingCartContext } from "../../utils/context/ShoppingCartContext";
+import { ShoppingStateEnums } from "../../utils/reducers/ShoppingCartReducer";
+import ReactModal from "react-modal";
+import { Link } from "react-router-dom";
+
+const ShoppingControls = ({ cartItem }) => {
+  const [quantity, setQuantity] = useState(0);
+  const [shoppingCartState, dispatchShoppingCart] =
+    useContext(ShoppingCartContext);
```

Comment:
My recommendation is to move this context outside of this component and leave it as a representational component, the context could be in the parent component, so in that way, the unit testing is simpler and cleaner.

Reply:
done